src/main/scala/io/iohk/ethereum/mpt/MerklePatriciaTrie.scala

```diff
@@ -241,37 +236,37 @@ class MerklePatriciaTrie[K, V] private (private val rootHash: Option[Array[Byte]
   }
 
   @tailrec
-  private def get(node: Node, searchKey: Array[Byte]): Option[Array[Byte]] = node match {
-    case LeafNode(key, value, _) =>
-      if (key sameElements searchKey) Some(value) else None
-    case extNode@ExtensionNode(sharedKey, _, _) =>
+  private def get(node: MptNode, searchKey: Array[Byte]): Option[Array[Byte]] = node match {
```

Comment:
I did not want to change mpt implementation too much, maybe we can do this refactoring in separate PR?